pkg/operational/metrics.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package operational
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+type MetricDefinition struct {
+	Name   string
+	Help   string
+	Type   metricType
+	Labels []string
+}
```

Comment:
@jotak IIUC, `Type metricType` is only used by the auto-documentation.
So, theoretically, one can create a `MetricDefinition` of type `Counter` and pass it to `NewGauge()` causing a misleading documentation.
If I'm correct, is it worth adding a check on `NewGauge()/NewCounter()/NewHistogram()` to validate the type?

Reply:
that's correct, sure we can do what you suggest